content/docs/user-guide/dvc-files-and-directories.md

```diff
@@ -73,6 +73,9 @@ An _output entry_ (`outs`) can have these fields:
   and a special _checksum_ for HDFS and WebHDFS.
 - `size`: Size of the file or directory (sum of all files).
 - `nfiles`: If a directory, number of files inside.
+- `isexec`: If a file, whether or not owner has execute permission. On Windows,
+  it is not saved and, if set manually, won't have any effect on `dvc checkout`
+  and `dvc pull`.
```

Comment:
```suggestion
- `isexec`: Whether this is an executable file. DVC preserves execute
  permissions upon `dvc checkout` and `dvc pull`. This has no effect on
  directories, or in general on Windows.
```

(same for `deps`)

Reply:
Questions:

- Do we want to specify "preserves POSIX execute permissions"? (possibly [linked](https://en.wikipedia.org/wiki/File-system_permissions#Traditional_Unix_permissions))
  TBH I see no need since we say it doesn't work on Windows so it's easy to infer why.
- "upon `dvc checkout` and `dvc pull`" - are these the only commands it affects? What about `add` or `commit` (when tracking stuff which moves it to the cache and then links back @efiop?

Thanks

Reply:
> Do we want to specify "preserves POSIX execute permissions"? (possibly linked)
TBH I see no need since we say it doesn't work on Windows so it's easy to infer why.

Thought about it too, but I feel it is generally understood as is, so I wouldn't clarify it more.

> "upon dvc checkout and dvc pull" - are these the only commands it affects? What about add or commit (when tracking stuff which moves it to the cache and then links back @efiop?

Pretty much yes. `add/commit` take existing file and add/commit it, so on windows there is nothing to capture - no exec bit to set. `checkout/pull` is when the exec bit is coming from another machine (e.g. your collegue on linux) or set by hand, and thats where it is better to clarify it.

Reply:
Hmm. I just tried this on Ubuntu (WSL) and DVC (1.11.8) removes the exec bit upon `add`. No reflinks on my system though so it's a copy, but the file doesn't have exec on either.

Reply:
\* but the cached file

Reply:
@jorgeorpinel 1.11.8 doesn't have that patch yet and it is expected for it to not preserve exc. You need to try upstream dvc.

Reply:
Sure but my question was: since add does remove the exec bit so, will the `exec` field affect add after all? Thinking about it that doesn't make sense as there is no .dvc file prior to add/commit so I'm re-approving this, but will there be perhaps a new flag on those options to do this?

Reply:
\*on those commands

Reply:
@jorgeorpinel Do you mean a flag to set isexec value through CLI? Not really, there is no need.

Reply:
p.s. Nvmd here. I asked the Q in the OP instead.

Reply:
Oh I see you answered here. OK